File: 2/2000s/21s/2138/DivideAStringIntoGroupsOfSizeK.cpp
/*
https://leetcode.com/problems/divide-a-string-into-groups-of-size-k/description/?envType=daily-question&envId=2025-06-22
*/

class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string>res;
        int sz=s.size(), idx=0;
        string curr="";
        while(true){
            if(idx>=sz) break;
            if(curr.size()==k){
                res.push_back(curr);
                curr="";
            } else{
                curr+=s[idx++];
            }
        }
        while(true){
            if(curr.size()==0||curr.size()==k) break;
            curr+=fill;
        }
        if(curr.size()>=1) res.push_back(curr);
        return res;
    }
};

File: 2/2000s/21s/2138/DivideAStringIntoGroupsOfSizeK.java
/*
https://leetcode.com/problems/divide-a-string-into-groups-of-size-k/description/?envType=daily-question&envId=2025-06-22
*/

class Solution {
    public String[] divideString(String s, int k, char fill) {
        ArrayList<String>resultantList=new ArrayList<>();
        int lengthOfS=s.length(), beginIndex=0;
        StringBuilder currentString=new StringBuilder();
        while(true){
            if(beginIndex>=lengthOfS) break;
            if(currentString.length()==k){
                resultantList.add(String.valueOf(currentString));
                currentString=new StringBuilder();
            } else{
                currentString.append(s.charAt(beginIndex));
                beginIndex++;
            }
        }
        while(true){
            if(currentString.length()==0||currentString.length()==k) break;
            currentString.append(fill);
        }
        if(currentString.length()>=1){
            resultantList.add(String.valueOf(currentString));
        }
        String[] resultantArray=new String[resultantList.size()];
        for(int i=0; i<resultantList.size(); ++i){
            resultantArray[i]=resultantList.get(i);
        }
        return resultantArray;
    }
}